From 98655214a3cb6dc0a0ee46f17ed5a0fcfa8ad684 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 14:58:09 -0400
Subject: [infiniband] usnic: Port over sysfs to new usnic_fwd.h

Message-id: <70415c3c0807c15edac5c7e0992cd0c3b67c3dd3.1411569661.git.dledford@redhat.com>
Patchwork-id: 96578
O-Subject: [Patch RHEL7 07/29] IB/usnic: Port over sysfs to new usnic_fwd.h
Bugzilla: 916382
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Upinder Malhi <umalhi@cisco.com>

This patch ports usnic_ib_sysfs.c to the new interface of
usnic_fwd.h.

Signed-off-by: Upinder Malhi <umalhi@cisco.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 60b215e8b267f911751a043de63181dab1b69706)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/usnic/usnic_ib_sysfs.c b/drivers/infiniband/hw/usnic/usnic_ib_sysfs.c
index bad985e..3e58842 100644
--- a/drivers/infiniband/hw/usnic/usnic_ib_sysfs.c
+++ b/drivers/infiniband/hw/usnic/usnic_ib_sysfs.c
@@ -104,7 +104,7 @@ usnic_ib_show_config(struct device *device, struct device_attribute *attr,
    PCI_SLOT(us_ibdev->pdev->devfn),
    PCI_FUNC(us_ibdev->pdev->devfn),
    netdev_name(us_ibdev->netdev),
-   us_ibdev->mac,
+   us_ibdev->ufdev->mac,
    atomic_read(&us_ibdev->vf_cnt.refcount));
   UPDATE_PTR_LEFT(n, ptr, left);
 
@@ -239,21 +239,17 @@ static ssize_t summary_show(struct usnic_ib_qp_grp *qp_grp, char *buf)
  int left;
  char *ptr;
  struct usnic_vnic_res_chunk *res_chunk;
- struct usnic_fwd_filter_hndl *default_filter_hndl;
  struct usnic_vnic_res *vnic_res;
 
  left = PAGE_SIZE;
  ptr = buf;
- default_filter_hndl = list_first_entry(&qp_grp->filter_hndls,
-     struct usnic_fwd_filter_hndl, link);
 
  n = scnprintf(ptr, left,
-   "QPN: %d State: (%s) PID: %u VF Idx: %hu Filter ID: 0x%x ",
+   "QPN: %d State: (%s) PID: %u VF Idx: %hu ",
    qp_grp->ibqp.qp_num,
    usnic_ib_qp_grp_state_to_string(qp_grp->state),
    qp_grp->owner_pid,
-   usnic_vnic_get_index(qp_grp->vf->vnic),
-   default_filter_hndl->id);
+   usnic_vnic_get_index(qp_grp->vf->vnic));
  UPDATE_PTR_LEFT(n, ptr, left);
 
  for (i = 0; qp_grp->res_chunk_list[i]; i++) {
-- 
1.7.1